example/src/components/RoomParticipant.tsx

```diff
@@ -18,23 +18,32 @@ import { PinButton } from './buttons/PinButton';
 
 type RoomParticipantProps = {
   participant: Membrane.Participant;
+  trackId?: string;
   onPinButtonPressed?: (string) => void;
   focused?: boolean;
   pinButtonHiddden?: boolean;
   tileSmall?: boolean;
 };
 
 export const RoomParticipant = ({
-  participant: { metadata, tracks, type, id },
+  participant,
+  trackId,
   onPinButtonPressed = (string) => {},
   focused = false,
   pinButtonHiddden = false,
   tileSmall = false,
 }: RoomParticipantProps) => {
+  const { metadata, tracks, type } = participant;
+
   const [showPinButton, setShowPinButton] = useState(false);
   const isPinButtonShown = useRef(false);
-  const videoTrack = tracks.find((t) => t.type === 'Video');
-  const audioTrack = tracks.find((t) => t.type === 'Audio');
+
+  const videoTrack = trackId ? tracks.find((t) => t.id === trackId) : null;
+  const videoTrackType = videoTrack?.metadata.type;
+  const audioTrack =
+    videoTrackType !== 'screensharing'
```

Comment:
I still think this check is unecessary (though it's simpler). You literally have the same one lower in the code (`videoTrackType !== 'screensharing' &&`)